pkg/reconciler/route/resources/ingress.go

```diff
@@ -232,18 +233,20 @@ func routeDomain(ctx context.Context, targetName string, r *servingv1.Route, vis
 	if isClusterLocal {
 		domains = ingress.ExpandedHosts(sets.NewString(domains...)).List()
 	}
-	return domains, err
+	return sets.NewString(domains...), err
 }
 
 // MakeACMEIngressPaths returns a set of netv1alpha1.HTTPIngressPath
 // that can be used to perform ACME challenges.
```

Comment:
should there be a comment added here to explain the new return value?

Reply:
I guess I was thinking something to explain what the extra domains are for. "Also returns hosts in the challenge that are not in domains". But I suppose that is just explaining the code at that point

Reply:
>  "Also returns hosts in the challenge that are not in domains"

This function is actually all the route domains (not the HTTP01 challenge domains)

ie.
`route-1.blah.example.com`
`route-1.blah.svc.cluster.local`
`route-1.blah.svc`
`route-1.blah`